sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -875,11 +878,20 @@ object DateTimeUtils {
         val hh = seconds / 3600
         val mm = seconds / 60 % 60
         val ss = seconds % 60
-        val nano = millisOfDay % 1000 * 1000000
+        val millis = millisOfDay % 1000
+        // Choose calendar based on java.util.Date getCalendarSystem
+        val calendar = if (year >= 1582) {
+          CalendarSystem.getGregorianCalendar()
```

Comment:
For this to happen millisLocal would have to be negative... which is at least well defined, but, would certainly need to be defined relative to January 1 1970 in the Gregorian calendar?